### DISCUSSION TITLE
Not Blank validator

### DISCUSSION THREAD

**Discussion by mrguamos** [danielgtaylor/huma#669] [Q&A] [answered]
How do you validate blank string? I can achieve this using resolver but is there any other way? I know there is minLength but it does not validate whitespace. With [go-validator](https://github.com/go-playground/validator) I can use [notblank](https://github.com/go-playground/validator/blob/master/non-standard/validators/notblank.go)

**Comment by danielgtaylor** [comment, marked as answer]
@mrguamos you should be able to use the JSON Schema `pattern` field for this, which works for both params and body fields:

```go
huma.Get(api, "/demo", func(ctx context.Context, input *struct {
	NotBlank string `query:"notblank" pattern:"^\\S+$"`
}) (*Response, error) { /* ... */ })
```

https://go.dev/play/p/u-XXqr4iF_u

Hope that helps!

**Reply by mrguamos** [reply]
ohh yeah, I can use pattern, why didn't I think of that. thank you!
